### ISSUE TITLE
Add support for swift 6 nightlies

### ISSUE THREAD

**Issue by mman** [swift-actions/setup-swift#673]
With the upcoming migration to Swift 6 and Swift Testing it would be cool if support for swift 6 can be added early via nightlies so that work in progress towards swift 6 migration, especially around concurrency and testing, can already be part of the workflows on GitHub.

**Comment by fwal** [comment]
Yeah I agree this would be great! The only real hurdle is resolving the path to the pkg to download as they contain the date when the last snapshot where created (unlike stable releases that have a fixed path).
Once that is figured out (would be great if they would have a feed or something) this would be easy to implement. 🙂

**Comment by mman** [comment]
Exactly, I'm aware of the limitation, I think it may need a collaboration from the swift build team to perhaps provide a stable symlink and or sum file so that the action can easily compare whether new nightly was published. Alternatively hardcoding the path with date to the action and re-release a patch version of this action automatically whenever new nightly is published, but that's I assume needles work unless automated as much as possible.

**Comment by khlopko** [comment]
Hi! [Swiftly](https://github.com/swiftlang/swiftly) fetches GitHub tags and uses regex to match specified snapshot and fetch date in the end ([parsing snapshot function](https://github.com/swiftlang/swiftly/blob/main/Sources/SwiftlyCore/HTTPClient%2BGitHubAPI.swift#L139)). Maybe this can be used to add snapshot support to GitHub Actions similarly as Swiftly does, and specified something like `swift-version: "main-snapshot"`.

**Comment by fwal** [comment]
As v3 of this action now have migrated to use Swiftly itself, nightlies and trunk should be fully supported 🎉
